modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/deployable/AbstractDeployableArtifactDescriptorFactory.java

```diff
@@ -0,0 +1,144 @@
+ */
+public abstract class AbstractDeployableArtifactDescriptorFactory<M extends MuleDeployableModel, T extends DeployableArtifactDescriptor>
+    extends AbstractArtifactDescriptorFactory<M, T> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractDeployableArtifactDescriptorFactory.class);
+
+  protected final DeployableProjectModel<M> deployableProjectModel;
+  protected final Optional<Properties> deploymentProperties;
+  private final PluginModelResolver pluginModelResolver;
+  private final PluginDescriptorResolver pluginDescriptorResolver;
+  private final ArtifactDescriptorFactory artifactDescriptorFactory;
+
+  public AbstractDeployableArtifactDescriptorFactory(DeployableProjectModel<M> deployableProjectModel,
+                                                     Map<String, String> deploymentProperties,
+                                                     PluginModelResolver pluginModelResolver,
+                                                     PluginDescriptorResolver pluginDescriptorResolver,
+                                                     ArtifactDescriptorValidatorBuilder artifactDescriptorValidatorBuilder,
+                                                     ArtifactDescriptorFactory artifactDescriptorFactory) {
+    super(deployableProjectModel.getProjectFolder(), deployableProjectModel.getMuleDeployableModel(),
+          artifactDescriptorValidatorBuilder);
+    this.deployableProjectModel = deployableProjectModel;
+    this.deploymentProperties = getProperties(deploymentProperties);
+    this.pluginModelResolver = pluginModelResolver;
+    this.pluginDescriptorResolver = pluginDescriptorResolver;
+    this.artifactDescriptorFactory = artifactDescriptorFactory;
+  }
+
+  private Optional<Properties> getProperties(Map<String, String> deploymentProperties) {
+    Properties properties = new Properties();
+    properties.putAll(deploymentProperties);
+    return of(properties);
+  }
+
+  @Override
+  protected ClassLoaderModel getClassLoaderModel() {
+    return new DeployableClassLoaderConfigurationAssembler<>(deployableProjectModel)
+        .createClassLoaderModel();
+  }
+
+  @Override
+  protected BundleDescriptor getBundleDescriptor() {
+    return deployableProjectModel.getBundleDescriptor();
+  }
+
+  @Override
+  protected void doDescriptorConfig(T descriptor) {
+    descriptor.setArtifactLocation(artifactLocation);
+    descriptor.setRedeploymentEnabled(artifactModel.isRedeploymentEnabled());
+
+    Set<String> configs = artifactModel.getConfigs();
+    if (configs != null && !configs.isEmpty()) {
+      descriptor.setConfigResources(new HashSet<>(configs));
+    } else {
+      descriptor.setConfigResources(ImmutableSet.<String>builder().add(getDefaultConfigurationResource()).build());
+    }
+
+    descriptor.setPlugins(createArtifactPluginDescriptors(descriptor));
+
+    // TODO: add log config file to descriptor, which currently relies in the mule home folder
```

Comment:
yup

Reply:
Created user story [W-11202321](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000xnuiQYAQ/view) to tackle it.